client/client.cpp

```diff
@@ -0,0 +1,168 @@
+#include "client.h"
+#include <QtWidgets>
+#include <QtNetwork>
+#include <QDebug>
+
+Client::Client(QWidget *parent)
+    : QDialog(parent)
+    ,hostCombo (new QComboBox)
+    ,portLineEdit(new QLineEdit)
+    ,getButton(new QPushButton(tr("Get ")))
+    , tcpSocket(new QTcpSocket(this))
+{
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+    hostCombo->setEditable(true);
+    QString name = QHostInfo::localHostName();
+    if (!name.isEmpty())
+    {
+        hostCombo->addItem(name);
+        QString domain = QHostInfo::localDomainName();
+
+        if (!domain.isEmpty())
+        {
+            hostCombo->addItem(name + QChar('.') + domain);
+        }
+    }
+    if (name != QLatin1String("localhost"))
+    {
+        hostCombo->addItem(QString("localhost"));
+    }
+    QList<QHostAddress> ipAddressesList = QNetworkInterface::allAddresses();
+
+    for(int i = 0; i < ipAddressesList.size(); ++i)
+    {
+        if (!ipAddressesList.at(i).isLoopback())
+            hostCombo->addItem(ipAddressesList.at(i).toString());
+    }
+    for (int i = 0; i < ipAddressesList.size(); ++i)
+    {
+        if (!ipAddressesList.at(i).isLoopback())
+            hostCombo->addItem(ipAddressesList.at(i).toString());
+    }
+
+    portLineEdit->setValidator(new QIntValidator(1, 65535, this));
+
+    auto hostLabel = new QLabel(tr("&Server name:"));
+    hostLabel->setBuddy(hostCombo);
+    auto portLabel = new QLabel(tr("&Server port:"));
+    portLabel->setBuddy(portLineEdit);
+
+    statusLabel = new QLabel(tr("This examples requires that you run the "
+                                " Server example as well."));
+    getButton->setDefault(false);
+    getButton->setEnabled(true);
+    auto quitButton = new QPushButton(tr("Quit"));
+
+    auto buttonBox = new QDialogButtonBox;
+    buttonBox->addButton(getButton, QDialogButtonBox::ActionRole);
+    buttonBox->addButton(quitButton, QDialogButtonBox::RejectRole);
+    in.setDevice(tcpSocket);
+    in.setVersion(QDataStream::Qt_4_0);
+    connect(hostCombo, &QComboBox::editTextChanged,
+            this, &Client::enableGetButton);
+    connect(portLineEdit, &QLineEdit::textChanged,
+            this, &Client::enableGetButton);
+    connect(getButton, &QAbstractButton::clicked,
```

Comment:
As discussed we don't want the client to request the data one by one, so we need either a button to just connect to a server, not to request a new line, maybe naming fixes could cover this